### ISSUE TITLE
Rule jsx-a11y/media-has-caption fails when it's conditional

### ISSUE THREAD

**Issue by binarykitchen** [jsx-eslint/eslint-plugin-jsx-a11y#1074]
Nice library, although I'm stuck with this one rule:
```
Media elements such as <audio> and <video> must have a <track> for captions.
```

As you can see in my code, it's conditional. When muted, then don't add VTT. Somehow, your plugin can't detect this and produces a false positive.

```tsx
<video
  className="replay"
  crossOrigin="anonymous"
  autoPlay
  preload="auto"
  playsInline
  controls
  poster={videomail.poster}
  muted={!hasAudio}
  style={{ width: videomail.width, height: videomail.height }}
>
  {videomail.mp4 && <source src={videomail.mp4} type="video/mp4" />}
  {videomail.webm && <source src={videomail.webm} type="video/webm" />}
  {hasAudio && <track src={videomail.vtt} kind="captions" default />}
</video>
```

Using all the latest versions here.

Thanks for investigating this.

**Comment by binarykitchen** [comment]
False alarm, my ESLint configuration was duplicate. Sorry.

**Comment by binarykitchen** [comment]
Hmmm, it still happens after I've fixed EsLint. Can we reopen it please?
